ReactWindows/ReactNative.Shared/Views/View/ReactViewManager.cs

```diff
@@ -322,18 +179,10 @@ public void SetBackgroundColor(BorderedCanvas view, uint color)
         [ReactProp("borderColor", CustomType = "Color")]
         public void SetBorderColor(BorderedCanvas view, uint? color)
         {
-            if (view.Border == null)
-            {
-                GetOrCreateBorderProps(view).Color = color;
-            }
-            else
-            {
-                var border = GetOrCreateBorder(view);
-                border.BorderBrush = color.HasValue
-                    ? new SolidColorBrush(ColorHelpers.Parse(color.Value))
-                    : DefaultBorderBrush;
-                TransferBackgroundBrush(view);
-            }
+            var xamlColor = color.HasValue
+                ? ColorHelpers.Parse(color.Value)
+                : (Color?)null;
+            view.SetBorderColor(xamlColor);
         }
```

Comment:
For some of these, would it also make sense to make these feel a bit more "C#-y", e.g., by making them properties?

Reply:
They would be write-only properties then, which is not very idiomatic in my opinion. And they will also "blend" with existing property-based standard API of Canvas, which would make it a little bit confusing for users. Are you sure you want them all as write-only properties?

---
In reply to: [177605056](https://github.com/Microsoft/react-native-windows/pull/1730#discussion_r177605056) [](ancestors = 177605056)

Reply:
Looking at:
- `SetBorderColor`
- `SetBackgroundColor`
- `SetBorderWidth`

---
In reply to: [177605056](https://github.com/Microsoft/react-native-windows/pull/1730#discussion_r177605056) [](ancestors = 177605056)

Reply:
Ah - okay. Yeah wasn't thinking about the fact that you'd need to "clobber" the Canvas proeprties.

---
In reply to: [177605762](https://github.com/Microsoft/react-native-windows/pull/1730#discussion_r177605762) [](ancestors = 177605762,177605056)

Reply:
I'm okay with some `new` properties clobbering the Canvas ones. They would have to be `write-only` because we don't currently have read implementations, correct?

---
In reply to: [177606484](https://github.com/Microsoft/react-native-windows/pull/1730#discussion_r177606484) [](ancestors = 177606484,177605762,177605056)

Reply:
The new write-only ones would not override/"new" existing ones but instead will bring more similar-looking and similar-meaning ones. Like now we have standard Background and I added BackgroundBrush which is already confusing. But if we convert all to properties it will add BackgroundColor making the whole API surface even more confusing. But I agree at least it would make it more uniform... Okay, let me redo the method API of BorderedCanvas to properties...

As to doing setters too, I think if they are not used and there are no plans for them to be used, we should not add them.

---
In reply to: [177606630](https://github.com/Microsoft/react-native-windows/pull/1730#discussion_r177606630) [](ancestors = 177606630,177606484,177605762,177605056)

Reply:
Why not "new" over existing properties? We should avoid confusion when possible, e.g., there should be only one way to set `Background`, `BorderWidth`, etc...

---
In reply to: [177607676](https://github.com/Microsoft/react-native-windows/pull/1730#discussion_r177607676) [](ancestors = 177607676,177606630,177606484,177605762,177605056)